### PR TITLE
feat(pt): add AdaMuon optimizer

### PR DESCRIPTION
**Pull request by OutisLi**
## Summary by CodeRabbit

* **New Features**
  * Added AdaMuon optimizer for PyTorch training with mixed 2D/1D parameter handling, batched orthogonalized updates, momentum and per-parameter state.

* **Configuration**
  * New optimizer options: momentum, weight decay, Adam betas, lr adjustment modes and tuning coefficients; AdaMuon selectable as an optimizer.

* **Tests**
  * Added comprehensive tests for orthogonalization, optimizer updates, bucketing, lr-adjust modes, weight decay, closure behavior, and state save/load.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: source/tests/pt/test_adamuon.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+                    self.assertTrue(
+                        torch.allclose(s1[key], s2[key]),
+                        f"State mismatch for key {key}",
+                    )
+                else:
+                    self.assertEqual(s1[key], s2[key], f"State mismatch for key {key}")
+
+
+if __name__ == "__main__":
+    unittest.main()
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply source/tests/pt/test_adamuon.py:387]: no way to set steps from json, use 5 forever

File: deepmd/pt/optimizer/adamuon.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+                m, n = rows, cols
+
+                # === Pre-compute bucket-level scaling constants ===
+                # RMS-aligned scale: 0.2 * sqrt(m * n)
+                rms_scale = 0.2 * math.sqrt(float(m * n))
+                # Shape-dependent lr correction (based on lr_adjust mode)
+                if lr_adjust <= 0:
+                    adj_scale = lr_adjust_coeff * math.sqrt(float(max(m, n)))
+                else:
+                    adj_scale = max(1.0, m / n) ** 0.5
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply deepmd/pt/optimizer/adamuon.py:369]: lr_adjust <= 0：使用 lr_adjust_coeff 控制 scaling，Adam 用 lr
lr_adjust > 0：使用 aspect ratio scaling，Adam 用 lr/lr_adjust
这是二元设计，不需要区分 == 0 和 < 0

File: deepmd/utils/argcheck.py [hunk 3]
```diff
@@ -3370,6 +3370,64 @@ def training_args(
...
+                        ),
+                        Argument(
+                            "lr_adjust_coeff",
+                            float,
+                            optional=True,
+                            default=0.2,
+                            doc=doc_only_pt_supported
+                            + "Coefficient for match-RMS scaling. Only effective when lr_adjust <= 0.",
+                        ),
+                    ],
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply deepmd/utils/argcheck.py:3427]: do not need to set

File: deepmd/pt/optimizer/adamuon.py [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+            X = torch.addmm(X, gram_update, X, beta=a, alpha=1.0)
+        else:
+            gram_update = torch.baddbmm(A, A, A, beta=b, alpha=c)
+            X = torch.baddbmm(X, gram_update, X, beta=a, alpha=1.0)
+
+    # === Step 6. Transpose back if needed ===
+    if G.size(-2) > G.size(-1):
+        X = X.mT
+
+    return X
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply deepmd/pt/optimizer/adamuon.py:109]: All implementation used these params

File: deepmd/pt/optimizer/adamuon.py [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply deepmd/pt/optimizer/adamuon.py:113]: mimic pytorch implementation
